pkg/activator/net/revision_backends.go

```diff
@@ -119,12 +107,98 @@ type destHealth struct {
 	health bool
 }
 
+func (rw *revisionWatcher) hasHealthyClusterIP() bool {
+	return rw.healthyClusterDest != ""
+}
+
+func (rw *revisionWatcher) revisionReady() bool {
+	return len(rw.dests) > 0
+}
+
+func (rw *revisionWatcher) getK8sPrivateService() (*corev1.Service, error) {
+	selector := labels.SelectorFromSet(map[string]string{
+		serving.RevisionLabelKey:  rw.rev.Name,
+		networking.ServiceTypeKey: string(networking.ServiceTypePrivate),
+	})
+	svcList, err := rw.serviceLister.Services(rw.rev.Namespace).List(selector)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(svcList) == 0 {
+		return nil, fmt.Errorf("Found no private service for revision %v", rw.rev)
+	}
+
+	if len(svcList) > 1 {
+		return nil, fmt.Errorf("Found multiple private services matching revision %v", rw.rev)
+	}
+
+	return svcList[0], nil
+}
+
+func (rw *revisionWatcher) probeClusterIP(svc *corev1.Service) (bool, string, error) {
+
+	svcPort, ok := GetServicePort(networking.ProtocolHTTP1, svc)
+	if !ok {
+		return false, "", fmt.Errorf("Unable to find HTTP1 port in service %s/%s", svc.Namespace, svc.Name)
+	}
+
+	// Context used for our probe requests
+	ctx, cancel := context.WithTimeout(context.Background(), probeTimeout)
+	defer cancel()
+
+	dest := net.JoinHostPort(svc.Spec.ClusterIP, strconv.Itoa(svcPort))
+	httpDest := url.URL{
+		Scheme: "http",
+		Host:   dest,
+	}
+	ok, err := prober.Do(ctx, rw.transport, httpDest.String(),
+		prober.WithHeader(network.ProbeHeaderName, queue.Name),
+		prober.ExpectsBody(queue.Name))
+
+	return ok, dest, err
+}
+
 // checkDests performs probing and potentially sends a dests update. It is
 // assumed this method is not called concurrently.
 func (rw *revisionWatcher) checkDests() {
 	healthStatesCh := make(chan destHealth, len(rw.dests))
 
-	// Context used for our requests
+	if rw.hasHealthyClusterIP() {
+		if !rw.revisionReady() {
+			// we have a healthy clusterIP but revision is not ready. We must have scaled down.
+			rw.healthyClusterDest = ""
+
+			// Send update that were now inactive
+			rw.updateCh <- &RevisionDestsUpdate{Rev: rw.rev}
+		}
+
+		// We had a healthy clusterIP so no need to check anything more
+		return
+	}
+
+	// First check the clusterIP
+	svc, err := rw.getK8sPrivateService()
+	if err != nil {
+		rw.logger.Errorw("Failed to lookup private service for revision", zap.Error(err))
+		return
+	}
+
+	if svc.Spec.ClusterIP == "" {
+		rw.logger.Errorf("Private service %s/%s clusterIP is nil, this should never happen", svc.ObjectMeta.Namespace, svc.ObjectMeta.Name)
+		return
+	}
+
+	// If clusterIP is healthy send this update and were done
+	if ok, dest, err := rw.probeClusterIP(svc); err != nil {
+		rw.logger.Errorw("Failed to probe clusterIP", zap.Error(err))
+	} else if ok {
+		rw.healthyClusterDest = dest
+		rw.updateCh <- &RevisionDestsUpdate{Rev: rw.rev, ClusterIPDest: dest}
```

Comment:
The caller is going to be the bookkeeping for request count tracking - it needs to know because podIPs need to limit to ContainerConcurrency and clusterIPs need to limit to ContainerConcurrency * replicaCnt.